Functions/Solution.cs

```diff
@@ -15,74 +16,267 @@ internal class Solution
+                        charInLetters = true;
+                        if (c + letterOffset > 'z')
+                            {
+                                newStr.Append((char)('a' + 'z' - c));
+                                break;
+                            }
+                        else
+                            newStr.Append((char)(c+letterOffset));
+                    }
+                }
+                if (!charInLetters)
+                    newStr.Append(c);
+            }
+            string changedStr = newStr.ToString();
+            Console.WriteLine(changedStr);
         }
 
         public static bool Assignment6(string str)
-        {
-            /*
-             * TODO: Write code here
-             */
+        {   
+            Regex rgx = new Regex("[^a-zA-Z]");
+            str = rgx.Replace(str, "");
+            string sentence = str.ToLower();
+            int n = sentence.Length - 1;
+            bool isPalindrome = true;
+            int i = 0;
+            while(i < n-i)
+            {
+                if (sentence[i] != sentence[n-i])
+                {
+                    isPalindrome = false;
+                    break;
+                }
+                i++;
+            }
+            return isPalindrome;
         }
 
         public static int Assignment7(int[] nums)
         {
-            /*
-             * TODO: Write code here
-             */
+            var histogram = new Dictionary<int, int>();
+            foreach(int num in nums)
+            {
+                if (histogram.ContainsKey(num))
+                    histogram[num]++;
+                else
+                    histogram[num] = 1;
+            }
+            foreach(KeyValuePair<int, int> entry in histogram)
+            {
+                if (entry.Value == 1)
+                    return entry.Key;
+            }
+            Console.WriteLine("No number in array with only 1 input");
+            return -1;
         }
 
+        
         public static void Assignment8(bool[,] board, int rows, int cols)
-        {
-            /*
-             * TODO: Write code here
-             */
+        {            
+            bool[,] newBoard = new bool[rows, cols];
+            for(int i = 0; i < rows; i++)
+            {
+                for (int j = 0; j < cols; j++)
+                {
+                    int numAliveNeighs = GameOfLife.calcAliveNeighs(board, i, j, rows, cols);
+                    if (board[i, j])
+                    {
+                        switch(numAliveNeighs)
+                        {
+                            case 0: case 1: case 4:
+                                newBoard[i, j] = false;
+                                break;
+                            case 2: case 3:
+                                newBoard[i, j] = true;
+                                break;
+                            default:
+                                Console.WriteLine("Shouldn't get here.");
```

Comment:
Then why not use `if` and `else if`?

Reply:
To save up "numAliveNeigbors == ..." a few times. Simply removed default dase.